sklearn/utils/estimator_checks.py

```diff
@@ -822,47 +833,10 @@ def check_estimators_overwrite_params(name, Estimator):
                      % (name, k, v, new_params[k]))
 
 
-def check_cluster_overwrite_params(name, Clustering):
-    X, y = make_blobs(random_state=0, n_samples=9)
-    with warnings.catch_warnings(record=True):
-        # catch deprecation warnings
-        clustering = Clustering()
-    set_fast_parameters(clustering)
-    params = clustering.get_params()
-    clustering.fit(X)
-    new_params = clustering.get_params()
-    for k, v in params.items():
-        assert_false(np.any(new_params[k] != v),
-                     "Estimator %s changes its parameter %s"
-                     " from %s to %s during fit."
-                     % (name, k, v, new_params[k]))
-
-
-def check_sparsify_multiclass_classifier(name, Classifier):
+def check_sparsify_coefficients(name, Estimator):
     X = np.array([[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1],
                   [-1, -2], [2, 2], [-2, -2]])
     y = [1, 1, 1, 2, 2, 2, 3, 3, 3]
-    est = Classifier()
-
-    est.fit(X, y)
-    pred_orig = est.predict(X)
-
-    # test sparsify with dense inputs
-    est.sparsify()
-    assert_true(sparse.issparse(est.coef_))
-    pred = est.predict(X)
-    assert_array_equal(pred, pred_orig)
-
-    # pickle and unpickle with sparse coef_
-    est = pickle.loads(pickle.dumps(est))
-    assert_true(sparse.issparse(est.coef_))
-    pred = est.predict(X)
-    assert_array_equal(pred, pred_orig)
```

Comment:
Actually my mistake, I was fooled by the folding of the github diff in the next test.